os/pm/pm_initialize.c

```diff
@@ -81,6 +81,26 @@ struct pm_global_s g_pmglobals;
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: pm_start
+ *
+ * Description:
+ *   This function is called by the application thread to start the Power
+ *   Management system. This fucntion sets the ready_to_state_change flag which
+ *   enables pm to transition between low and high power states.
+ *
+ * Input parameters:
+ *   None.
+ *
+ * Returned value:
+ *    None.
+ *
+ ****************************************************************************/
+
+void pm_start(void) {
+	g_pmglobals.ready_to_state_change = true;
```

Comment:
Even if we need this interface, cant we need to check this flag as well in the places where we do the state changes for the pm?

Reply:
the `pm_changestate` is the only function which do pm state transition, we have used this flag in it to stop state transition